File: ConduitAPI/src/main/java/systems/conduit/api/inventory/ChestContainer.java
package systems.conduit.api.inventory;

import lombok.Getter;
import net.minecraft.world.ContainerListener;
import net.minecraft.world.SimpleContainer;
import net.minecraft.world.inventory.ChestMenu;
import net.minecraft.world.inventory.MenuType;

public class ChestContainer extends SimpleContainer {

    @Getter private MenuType<ChestMenu> type;
    @Getter private String title = "";

    private ChestContainer(int size) {
        super(size);
    }

    public static ChestContainer create(MenuType<ChestMenu> size, String title) {
        ChestContainer chestContainer = new ChestContainer(getSizeFromEnum(size));
        chestContainer.type = size;
        chestContainer.title = title;
        return chestContainer;
    }

    // TODO: Use?
    @Override
    public void addListener(ContainerListener var1) {
        throw new UnsupportedOperationException("Container listener not implemented!");
    }

    private static int getSizeFromEnum(MenuType<ChestMenu> size) {
        if (MenuType.GENERIC_9x2.equals(size)) {
            return 18;
        } else if (MenuType.GENERIC_9x3.equals(size)) {
            return 27;
        } else if (MenuType.GENERIC_9x4.equals(size)) {
            return 36;
        } else if (MenuType.GENERIC_9x5.equals(size)) {
            return 45;
        } else if (MenuType.GENERIC_9x6.equals(size)) {
            return 54;
        }
        return 9;
    }
}

File: src/main/kotlin/systems/conduit/api/inventory/ChestContainer.kt
package systems.conduit.api.inventory

import net.minecraft.world.ContainerListener
import net.minecraft.world.SimpleContainer
import net.minecraft.world.inventory.ChestMenu
import net.minecraft.world.inventory.MenuType

class ChestContainer private constructor(size: Int): SimpleContainer(size) {

    var type: MenuType<ChestMenu>? = null


    var title = ""
        private set

    // TODO: Use?
    override fun addListener(var1: ContainerListener) {
        throw UnsupportedOperationException("Container listener not implemented!")
    }

    companion object {
        fun create(size: MenuType<ChestMenu>, title: String): ChestContainer {
            val chestContainer = ChestContainer(getSizeFromEnum(size))
            chestContainer.type = size
            chestContainer.title = title
            return chestContainer
        }

        private fun getSizeFromEnum(size: MenuType<ChestMenu>): Int {
            if (MenuType.GENERIC_9x2 == size) {
                return 18
            } else if (MenuType.GENERIC_9x3 == size) {
                return 27
            } else if (MenuType.GENERIC_9x4 == size) {
                return 36
            } else if (MenuType.GENERIC_9x5 == size) {
                return 45
            } else if (MenuType.GENERIC_9x6 == size) {
                return 54
            }
            return 9
        }
    }
}